SRC/list.js

```diff
@@ -0,0 +1,164 @@
+  const li = document.createElement('li');
+  li.id = id;
+  li.classList.add('list');
+  li.setAttribute('draggable', 'true');
+  if (completed) li.classList.add('checked');
+  const text = document.createElement('p');
+  text.classList.add('text');
+  const moreIcon = document.createElement('i');
+  const div = document.createElement('div');
+  text.innerHTML = description;
+  const checbox = document.createElement('input');
+  checbox.setAttribute('type', 'checkbox');
+  checbox.classList.add('checkbox');
+  moreIcon.setAttribute('class', 'fas fa-ellipsis-v');
+  checbox.checked = completed;
+  div.appendChild(checbox);
+  div.appendChild(text);
+  li.appendChild(div);
+  li.appendChild(moreIcon);
+  theList.appendChild(li);
+};
+
+const localStorageList = JSON.parse(localStorage.getItem('list'));
+
+const localRender = () => {
+  theList.innerHTML = '';
+  localStorageList
+    .sort((a, b) => a.index - b.index)
+    .forEach((item) => {
+      renderListElemenet(item.description, item.completed, item.id);
+    });
+
+  addActionListner();
+  if (localStorage > 0) {
+    addChecklocal();
+  } else {
+    addCheck();
+  }
+};
+
+const render = () => {
+  theList.innerHTML = '';
+  listItems
+    .sort((a, b) => a.index - b.index)
+    .forEach((item) => {
+      renderListElemenet(item.description, item.completed, item.id);
+    });
+
+  addActionListner();
+  addActionListner();
+  if (localStorage > 0) {
+    addChecklocal();
+  } else {
+    addCheck();
+  }
+};
+
+const swapLocal = (firstId, secondId) => {
+  localStorageList = JSON.parse(localStorage.getItem('list'));
+  const firstObject = localStorageList.find(
+    (ele) => ele.id === Number(firstId)
+  );
+  const secondObject = localStorageList.find(
+    (ele) => ele.id === Number(secondId)
+  );
+
+  const tempIndex = firstObject.index;
+
+  console.log(listItems);
+
+  const tempId = listItems[tempIndex].id;
+  const tempDesc = listItems[tempIndex].description;
+  const tempCompl = listItems[tempIndex].completed;
+  // let tempInd = listItems[firstId].index;
+
+  listItems[tempIndex].id = listItems[secondObject.index].id;
+  listItems[tempIndex].description = listItems[secondObject.index].description;
+  listItems[tempIndex].completed = listItems[secondObject.index].completed;
+  // listItems[firstId].index = listItems[secondId].index;
+
+  listItems[secondObject.index].id = tempId;
+  listItems[secondObject.index].description = tempDesc;
+  listItems[secondObject.index].completed = tempCompl;
+  // listItems[secondId].index = tempInd;
+
+  // debugger;
+  console.log(listItems);
+
+  localStorage.setItem('list', JSON.stringify(listItems));
+  localStorageList = JSON.parse(localStorage.getItem('list'));
+
+  firstObject.index = secondObject.index;
+  secondObject.index = tempIndex;
+
+  if (localStorage.length > 0) {
+    localRender();
+  } else {
+    render();
+  }
+};
+const swap = (firstId, secondId) => {
+  // localStorageList = JSON.parse(localStorage.getItem('list'));
```

Comment:
- [ ] [OPTIONAL] Remember to delete all commented code, clean code it's a happy code